### DISCUSSION TITLE
bot doesn't work

### DISCUSSION THREAD

**Discussion by twogog** [telegraf/telegraf#1864] [Q&A] [answered]
serverless function works fine (Vercel), but bot doesn't answer when POST mes reseives from telegram. I can't figure out a problem. 
https://github.com/twogog/tg-bot/blob/main/api/bot.js

update. Guess I'm just stupid trying to instanciate a bot whenever reseving POST messaging. It doesn't make sense...

**Comment by MKRhere** [comment, marked as answer]
Move all Telegraf code outside the exported handler. Remove bot.launch as well. You don't need it in serverless. You see here, you just need to pass the update into Telegraf using `handleUpdate` for it to handle it from there.

```TS
const bot = new Telegraf(process.env.TOKEN);
bot.on(message("text"), (ctx) => ctx.reply("Hello"));

module.exports = async (request, response) => {
  try {
	await bot.handleUpdate(request.body);
  } catch (error) {
    console.error("Error handling update", error.message);
  }

  response.send("OK");
};
```

Remember you need to tell Telegram Bot API separately about your webhook somehow (only once, unless the URL changes), for example using the telegraf cli:

```sh
npx telegraf -t $TOKEN -m setWebhook -D '{ "url": $FULL_URL_TO_DEPLOYED_FUNCTION }'
```

**Reply by twogog** [reply]
Thank you) Now it works
